src/utils.py

```diff
@@ -1261,29 +1416,40 @@ def benchmark(models, num_times, X, y, benchmark, train_size = 0.7, val_size = 0
         )
 
         #recombine train and val for methods that don't use a val set, and for training model after finding markers
-        X_train = X[np.concatenate([train_indices, val_indices]), :]
-        y_train = y[np.concatenate([train_indices, val_indices])]
+        X_train = X[train_indices, :]
+        y_train = y[train_indices]
+        X_val = X[val_indices, :]
+        y_val = y[val_indices]
+        X_train_val = X[np.concatenate([train_indices, val_indices]), :]
+        y_train_val = y[np.concatenate([train_indices, val_indices])]
         X_test = X[test_indices,:]
         y_test = y[test_indices]
 
```

Comment:
I think these comments need to be removed in the new stuff.

I am a bit confused about the train and validation stuff since some of the methods that we compare to don't make a distinction between train and validation sets when forming the models that are proxies to evaluate the markers.

Can you explain a bit this change?

Reply:
The comment that goes 

#recombine train and val for methods that don't use a val set, and for training model after finding markers

Reply:
Ah good catch, yeah that comment isn't accurate anymore. 

For the data splitting, I am glad you brought it up because it is a little wacky how I am doing it right now, so let me know if you think there are better ways of handling it. Basically the complication stems from the fact that I want to call the data splitting in the outermost loop so that all the models use the same data splits. But then the different models use different parts of the data.

So LassoNet uses X_train, y_train, X_val, and y_val. Then the VAE family of models use train_dataloader and val_dataloader. I haven't done the Smash wrapper yet, but that one is even worse because it needs the annotations. So here I was thinking, just pass all the stuff to the benchmarkerFunctional, and then in that function for each model, just use the ones you need.

The X_train_val and y_train_val is just passed to the new_model_metrics. I think I might move that into the function call to make it less confusing.

Reply:
Yeah RankCorr and SmashPy have the same issue. 


> and then in that function for each model, just use the ones you need.

Yeah by using some classmethod or a partial or wtv. I think that could work.

Sounds good to me!